File: frontend/src/services/adminService.js
import api from './api';

export const adminService = {
  // Dashboard
  getStats: () => api.get('/admin/dashboard/stats'),
  
  // Customers
  getCustomers: () => api.get('/admin/customers'),
  getCustomer: (ssn) => api.get(`/admin/customers/${ssn}`),
  createCustomer: (data) => api.post('/admin/customers', data),
  updateCustomer: (ssn, data) => api.put(`/admin/customers/${ssn}`, data),
  deleteCustomer: (ssn) => api.delete(`/admin/customers/${ssn}`),
  getCustomerCount: () => api.get('/admin/customers/count'),
  
  // Rooms
  getRooms: () => api.get('/admin/rooms'),
  getRoom: (number) => api.get(`/admin/rooms/${number}`),
  createRoom: (data) => api.post('/admin/rooms', data),
  updateRoom: (number, data) => api.put(`/admin/rooms/${number}`, data),
  deleteRoom: (number) => api.delete(`/admin/rooms/${number}`),
  
  // Room Types
  getRoomTypes: () => api.get('/room-types'),
  getRoomStatusTypes: () => api.get('/room-status-types'),
  
  // Reservations
  getReservations: () => api.get('/admin/reservations'),
  getReservationById: (id) => api.get(`/admin/reservations/${id}`),
  updateReservationStatus: (id, status) => 
    api.put(`/admin/reservations/${id}/status`, { status }),
  getReservationCount: () => api.get('/admin/reservations/count'),
  getTodayCheckins: () => api.get('/admin/reservations/today-checkins'),
  getTodayCheckouts: () => api.get('/admin/reservations/today-checkouts'),
  
  // Staff
  getStaff: () => api.get('/admin/staff'),
  getStaffMember: (ssn) => api.get(`/admin/staff/${ssn}`),
  createStaff: (data) => api.post('/admin/staff', data),
  updateStaff: (ssn, data) => api.put(`/admin/staff/${ssn}`, data),
  deleteStaff: (ssn) => api.delete(`/admin/staff/${ssn}`),
  getStaffCount: () => api.get('/admin/staff/count'),
  
  // Departments
  getDepartments: () => api.get('/admin/departments'),
  createDepartment: (data) => api.post('/admin/departments', data),
  updateDepartment: (id, data) => api.put(`/admin/departments/${id}`, data),
  deleteDepartment: (id) => api.delete(`/admin/departments/${id}`),
  
  // Services
  getAdminServices: () => api.get('/admin/services'),
  createService: (data) => api.post('/admin/services', data),
  updateService: (id, data) => api.put(`/admin/services/${id}`, data),
  deleteService: (id) => api.delete(`/admin/services/${id}`),
  
  // Discounts
  getDiscounts: () => api.get('/admin/discounts'),
  createDiscount: (data) => api.post('/admin/discounts', data),
  updateDiscount: (id, data) => api.put(`/admin/discounts/${id}`, data),
  deleteDiscount: (id) => api.delete(`/admin/discounts/${id}`),
  
  // Price List
  getPriceList: () => api.get('/admin/price-list'),
  createPriceEntry: (data) => api.post('/admin/price-list', data),
  updatePriceEntry: (id, data) => api.put(`/admin/price-list/${id}`, data),
  deletePriceEntry: (id) => api.delete(`/admin/price-list/${id}`),
  
  // Membership
  getMembershipLevels: () => api.get('/membership-levels'),
};

File: backend/src/main/java/com/hotel/service/AdminService.java
package com.hotel.service;

import com.hotel.dao.*;
import com.hotel.dto.DashboardStatsDTO;
import com.hotel.entity.*;
import java.sql.SQLException;
import java.util.List;

public class AdminService {
    
    private CustomerDAO customerDAO = new CustomerDAO();
    private RoomDAO roomDAO = new RoomDAO();
    private ReservationDAO reservationDAO = new ReservationDAO();
    private StaffDAO staffDAO = new StaffDAO();
    
    // Customer operations
    public List<Customer> getAllCustomers() throws SQLException {
        return customerDAO.findAll();
    }
    
    public Customer getCustomer(String ssn) throws SQLException {
        return customerDAO.findBySsn(ssn);
    }
    
    public long getCustomerCount() throws SQLException {
        return customerDAO.count();
    }
    
    public Customer createCustomer(Customer customer) throws SQLException {
        customerDAO.create(customer);
        return customer;
    }
    
    public Customer updateCustomer(Customer customer) throws SQLException {
        customerDAO.update(customer);
        return customer;
    }
    
    public void deleteCustomer(String ssn) throws SQLException {
        customerDAO.delete(ssn);
    }
    
    // Room operations
    public List<Room> getAllRooms() throws SQLException {
        return roomDAO.findAll();
    }
    
    public Room getRoom(Integer roomNumber) throws SQLException {
        return roomDAO.findByNumber(roomNumber);
    }
    
    public Room createRoom(Room room) throws SQLException {
        roomDAO.create(room);
        return room;
    }
    
    public Room updateRoom(Room room) throws SQLException {
        roomDAO.update(room);
        return room;
    }
    
    public void deleteRoom(Integer roomNumber) throws SQLException {
        roomDAO.delete(roomNumber);
    }
    
    // Reservation operations
    public List<Reservation> getAllReservations() throws SQLException {
        return reservationDAO.findAll();
    }
    
    public long getReservationCount() throws SQLException {
        return reservationDAO.count();
    }
    
    public long getTodayCheckins() throws SQLException {
        return reservationDAO.getTodayCheckins();
    }
    
    public long getTodayCheckouts() throws SQLException {
        return reservationDAO.getTodayCheckouts();
    }
    
    public Reservation updateReservationStatus(Integer id, String status) throws SQLException {
        reservationDAO.updateStatus(id, status);
        return reservationDAO.findById(id);
    }
    
    // Staff operations
    public List<Staff> getAllStaff() throws SQLException {
        return staffDAO.findAll();
    }
    
    public long getStaffCount() throws SQLException {
        return staffDAO.count();
    }
    
    public Staff createStaff(Staff staff) throws SQLException {
        staffDAO.create(staff);
        return staff;
    }
    
    public Staff updateStaff(Staff staff) throws SQLException {
        staffDAO.update(staff);
        return staff;
    }
    
    public void deleteStaff(String ssn) throws SQLException {
        staffDAO.delete(ssn);
    }
    
    // Service operations (placeholder)
    public List<Service> getAllServices() throws SQLException {
        // Return empty list for now - implement ServiceDAO if needed
        return List.of();
    }
    
    public Service createService(Service service) throws SQLException {
        return service;
    }
    
    public Service updateService(Service service) throws SQLException {
        return service;
    }
    
    public void deleteService(Integer id) throws SQLException {
        // Implement if needed
    }
    
    // Department operations (placeholder)
    public List<Department> getAllDepartments() throws SQLException {
        return List.of();
    }
    
    public Department createDepartment(Department department) throws SQLException {
        return department;
    }
    
    public Department updateDepartment(Department department) throws SQLException {
        return department;
    }
    
    public void deleteDepartment(String id) throws SQLException {
        // Implement if needed
    }
    
    // Discount operations (placeholder)
    public List<Discount> getAllDiscounts() throws SQLException {
        return List.of();
    }
    
    public Discount createDiscount(Discount discount) throws SQLException {
        return discount;
    }
    
    public Discount updateDiscount(Discount discount) throws SQLException {
        return discount;
    }
    
    public void deleteDiscount(Integer id) throws SQLException {
        // Implement if needed
    }
    
    // Price List operations (placeholder)
    public List<PriceList> getAllPriceList() throws SQLException {
        return List.of();
    }
    
    public PriceList createPriceEntry(PriceList priceEntry) throws SQLException {
        return priceEntry;
    }
    
    public PriceList updatePriceEntry(PriceList priceEntry) throws SQLException {
        return priceEntry;
    }
    
    public void deletePriceEntry(Integer listNo) throws SQLException {
        // Implement if needed
    }
    
    // Dashboard stats
    public DashboardStatsDTO getDashboardStats() throws SQLException {
        DashboardStatsDTO stats = new DashboardStatsDTO();
        stats.setTotalCustomers(customerDAO.count());
        stats.setTotalRooms(roomDAO.findAll().size());
        stats.setTotalReservations(reservationDAO.count());
        stats.setTotalStaff(staffDAO.count());
        stats.setTodayCheckins(reservationDAO.getTodayCheckins());
        stats.setTodayCheckouts(reservationDAO.getTodayCheckouts());
        
        // Calculate available and occupied rooms
        List<Room> rooms = roomDAO.findAll();
        long available = 0, occupied = 0;
        for (Room room : rooms) {
            if (room.getStatusId() != null && room.getStatusId() == 1) available++;
            else if (room.getStatusId() != null && room.getStatusId() == 2) occupied++;
        }
        stats.setAvailableRooms(available);
        stats.setOccupiedRooms(occupied);
        stats.setMonthlyRevenue(45890.00);
        
        return stats;
    }
}